Kzl 241 document controller (#86)
kuzzleio/documentation-V2: src/sdk-reference/cpp/1/document/create/index.md

Throws if a document with the same given id already exists in Kuzzle.

The optional parameter `refresh` can be used with the value `wait_for` in order to wait for the document to be indexed (indexed documents are available for `search`).

## Arguments

```cpp
std::string create(
    const std::string& index, 
    const std::string& collection, 
    const std::string& id, 
    const std::string& body, 
    kuzzleio::query_options *options=nullptr
);
```

| Argument | Type | Description |
| --- | --- | --- |
| `index` | <pre>std::string</pre> | Index name |
| `collection` | <pre>std::string</pre> | Collection name |
| `id` | <pre>std::string</pre> | Optional document id. If set to a blank string, will use a auto-generated id |
| `body` | <pre>std::string</pre> | A JSON string containing the body of the document |
| `options` | <pre>query_options</pre> | A pointer to a `query_options` containing query options |

### options

Additional query options

| Option   | Type (default)    | Description                       |
| ---------- | ------- | --------------------------------- |
| `queuable` | <pre>boolean</pre> (`true`) | Make this request queuable or not |

> | `refresh` | <pre>std::string</pre> | If set to `wait_for`, waits for the change to be reflected for `search` (up to 1s) |

Reviewer: ```suggestion
| `refresh` | <pre>std::string</pre> | If set to `wait_for`, responds once the change has been indexed (up to 1s) |
```

(if this suggestion is accepted, it should be copy/pasted to every other occurences of the `refresh` option documentation in this PR)

Author: I am not found of the proposal. A document being index does not tell the end user what is the impact on using the option or not.
The existing sentence is far being perfect but at least gives some input on what the refresh option can be useful to.
